### PR TITLE
Add Logger Service  and ExtensionService

### PR DESCRIPTION
**Pull request by Lordapa**

### CODE REVIEW

File: src/DonateTo.Infrastructure/DonateTo.Infrastructure.csproj [hunk 1]
```diff
@@ -6,14 +6,19 @@
...
   </ItemGroup>
 
   <ItemGroup>
     <PackageReference Include="Microsoft.CodeAnalysis.FxCopAnalyzers" Version="2.9.8">
       <PrivateAssets>all</PrivateAssets>
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
     </PackageReference>
+    <PackageReference Include="Microsoft.Extensions.Logging" Version="3.1.3" />
+    <PackageReference Include="Microsoft.Extensions.Logging.Console" Version="3.1.3" />
+    <PackageReference Include="Serilog" Version="2.9.1-dev-01154" />
```
Comment (nicolasgranata) [inline src/DonateTo.Infrastructure/DonateTo.Infrastructure.csproj:18]: Is this version -dev a pre-release?

File: src/DonateTo.Infrastructure/Logging/LoggingExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+using Serilog;
+using Serilog.Extensions.Logging;
+
+namespace DonateTo.Infrastructure.Logging
+{
+    public static class LoggingExtensions
+    {
+        public static IServiceCollection AddLoggingToPipeline(this IServiceCollection services, IConfiguration configuration)
+        {
+            var cs = configuration.GetConnectionString("PostgreSQL");
```
Comment (nicolasgranata) [inline src/DonateTo.Infrastructure/Logging/LoggingExtensions.cs:13]: Change the name of the variable cs is not enough descriptive

File: src/DonateTo.WebApi/Startup.cs [hunk 3]
```diff
@@ -39,7 +42,7 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env, IApiVers
 
             app.UseRouting();
 
-            app.UseSwaggerWithVersioning(provider);
+            app.UseSwaggerWithVersioning(provider, Configuration);
```
Comment (nicolasgranata) [inline src/DonateTo.WebApi/Startup.cs:45]: This is not required. Please pull it from develop.

File: src/DonateTo.WebApi/Swagger/SwaggerExtensions.cs [hunk 4]
```diff
@@ -34,8 +37,8 @@ public static IServiceCollection AddSwagger(this IServiceCollection services)
 
             return services;
         }
-
-        public static IApplicationBuilder UseSwaggerWithVersioning(this IApplicationBuilder app, IApiVersionDescriptionProvider provider)
+
+        public static IApplicationBuilder UseSwaggerWithVersioning(this IApplicationBuilder app, IApiVersionDescriptionProvider provider, IConfiguration configuration)
```
Comment (nicolasgranata) [inline src/DonateTo.WebApi/Swagger/SwaggerExtensions.cs:41]: This is not required. Please pull it from develop.

File: src/DonateTo.WebApi/V1/Controllers/SampleController.cs [hunk 5]
```diff
@@ -9,11 +12,14 @@ namespace DonateTo.WebApi.V1.Controllers
     [ApiController]
     public class SampleController : ControllerBase
     {
+
         [HttpGet]
         [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(IEnumerable<string>))]
         [ProducesResponseType(StatusCodes.Status500InternalServerError)]
         public ActionResult<IEnumerable<string>> Get()
         {
+            Logger.Debug("stasas");
```
Comment (nicolasgranata) [inline src/DonateTo.WebApi/V1/Controllers/SampleController.cs:21]: Change this to something more descriptive, such as "Testing logger" in the message.

File: src/DonateTo.Infrastructure/Logging/SerilogLogger.cs [hunk 6]
```diff
@@ -0,0 +1,28 @@
+﻿using Serilog;
+using System;
+
+namespace DonateTo.Infrastructure.Logging
+{
+    public sealed class Logger
```
Comment (nicolasgranata) [inline src/DonateTo.Infrastructure/Logging/SerilogLogger.cs:6]: Static classes are sealed if we are not going to add non-static methods, this class should be static

File: src/DonateTo.WebApi/Startup.cs (hunk 3 above)
Comment (Lordapa) [reply src/DonateTo.WebApi/Startup.cs:45]: this is required by the cto

File: src/DonateTo.Infrastructure/Logging/SerilogLogger.cs (hunk 6 above)
Comment (Lordapa) [reply src/DonateTo.Infrastructure/Logging/SerilogLogger.cs:6]: Also change the name of the file from SerilogLogger to Logger

**Pull request merged**